pkg/utils/utils_virtual.go

```diff
@@ -205,6 +226,33 @@ func getOpenstackDataFromMetadataService() (metaData *OSPMetaData, networkData *
 	return metaData, networkData, nil
 }
 
+// getPCIAddressFromMACAddress returns the PCI address of a device given its MAC address
+func getPCIAddressFromMACAddress(macAddress string) (string, error) {
+	nics, err := ghw.Network()
+	if err != nil {
+		return "", fmt.Errorf("error getting network info: %v", err)
+	}
+
+	var pciAddress string
+	for _, nic := range nics.NICs {
+		if nic.MacAddress == macAddress {
```

Comment:
I don't know how these are returned, but wouldn't it be safer to lowercase the MAC addresses (both `nic.MacAddress` and `macAddress`) before comparing?